File: mfd/src/pso/PartPop.scala
package pso

import fitness.MFDFitnessFunction

class PartPop(popSize:Int, dims: Int, c1: Double, c2: Double, vMax: Double, w: Double, fitFunc: MFDFitnessFunction) {

  var pop = Array.ofDim[Particle](popSize)
  pop(0) = new Particle(dims, c1, c2, vMax, w, fitFunc)
  var gbest = pop(0).x
  var gbestFit = pop(0).fitness

  for (i <- 1 until pop.length){
    pop(i) = new Particle(dims, c1, c2, vMax, w, fitFunc)
    if (pop(i).fitness > gbestFit){
      gbest = pop(i).x
      gbestFit = pop(i).fitness
    }
  }
}

File: pso_java/PartPop.java


public class PartPop {

    byte[] gBest; 
    double gBestFit; 
    Particle[] pop;

    public PartPop (int popSize, int dims, double c1, double c2,
                    double vMin, double vMax, double w, 
                    MFDFitnessFunction fitFunc) {
        pop = new Particle[popSize];
        pop[0] = new Particle (dims, c1, c2, vMin, vMax, w, fitFunc);
        gBest = pop[0].x;
        gBestFit = pop[0].fitness;

        for (int i = 0; i < pop.length; i++) {
            pop[i] = new Particle (dims, c1, c2, vMin, vMax, w, fitFunc); 
            if (pop[i].fitness > gBestFit) {
                gBest = pop[i].x;
                gBestFit = pop[i].fitness;
            }
        }
    }
}
